### PR TITLE
CircleCI 2.0

### PR DESCRIPTION
**Pull request by felicianotech**
Hey everyone,

If you don't know, CircleCI 1.0 [is being EOL'd](https://circleci.com/blog/sunsetting-1-0/) at the end of August. I noticed that this project uses CircleCI 1.0 and so in the spirit of open source I wanted to submit a PR to do the migration.

The **TL:DR;** of this PR is that it converts the CircleCI 1.0 configuration file into the equivalent CircleCI 2.0 configuration file. I'd say that the new config file is 99% the same software-wise. The version of Docker client is newer now (the one in CircleCI 1.0 is very old).

The average build time might drop a little bit as well. Rightnow the avg build time is approx 33 minutes. The CircleCI 2.0 version should be 3- 6 minutes less than that for a complete build.

I have tested this build myself, however my builds fail once it comes to uploading to AWS S3. As you can imagine, I don't have your credentials for AWS and Docker Hub so I was not able to test the build past that point.

Basically, this PR allows this project to migrate to CircleCI 2.0 before the August 2018 EOL of 1.0 as well as slightly decrease build times.

I'm happy to answer any questions or help with any tweaks to get this PR merged if you choose to do so.

  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by lukevmorris** [comment]
@felicianotech Great! I'll merge this now to test vs the master branch, and make any necessary tweaks there. Thank you so much for your contribution!
